sound/soc/sof/sof-client.c

```diff
@@ -610,3 +610,50 @@ enum sof_fw_state sof_client_get_fw_state(struct sof_client_dev *cdev)
 	return sdev->fw_state;
 }
 EXPORT_SYMBOL_NS_GPL(sof_client_get_fw_state, SND_SOC_SOF_CLIENT);
+
+int sof_client_load_firmware(struct sof_client_dev *cdev)
+{
+	struct snd_sof_dev *sdev = sof_client_dev_to_sof_dev(cdev);
+
+	return snd_sof_load_firmware(sdev);
+}
+EXPORT_SYMBOL_NS_GPL(sof_client_load_firmware, SND_SOC_SOF_CLIENT);
+
+void sof_client_unload_firmware(struct sof_client_dev *cdev)
+{
+	struct snd_sof_dev *sdev = sof_client_dev_to_sof_dev(cdev);
+
+	snd_sof_fw_unload(sdev);
+}
+EXPORT_SYMBOL_NS_GPL(sof_client_unload_firmware, SND_SOC_SOF_CLIENT);
```

Comment:
I'm not sure how this can work in practice...
The combination of the two is kind of forces to re-load the firmware from filesystem and at the same time it is also going to try to copy to DSP memory (`snd_sof_load_firmware_memcpy()` variant).